project-ideas_2023/README.md

```diff
@@ -0,0 +1,272 @@
+
+Apart from the goals above, the following general tasks are expected:
+
+- General understanding of REST APIs, Apache Kafka and the Kafka Connect architecture
+- Understand the association of classes and modules in the code with the components in the architecture.
+- Understand the specification provided by Fitbit Web API and design implementations based on that.
+
+**Skills:**
+
+- _Must have:_ Java, REST API
+- _Good-to-have:_ Docker, Kafka
+
+**Difficulty:** Medium
+
+**Expected Size:** 350-hour (Full time)
+
+**Mentors**: [yatharth](https://github.com/yatharthranjan)
+
+---
+
+### 4. Grafana Dashboards for visualising the data in realtime
+
+**Overview**: The platform uses [Grafana](https://grafana.com/) to visualise the data coming from the devices and apps. A TimescaleDB Kafka connector delivers both active and passive data from the Kafka topics to a TimescaleDB database. Data from the database is then used to visualise data completeness through various [dashboards in Grafana](https://github.com/RADAR-base/RADAR-Grafana). 
+
+![Screen Shot 2022-02-21 at 3 27 47 PM](https://user-images.githubusercontent.com/16977973/154984871-bce650ce-28ce-40f4-a558-dd60d79ce89d.png)
+
+**Current Status**: Currently, the dashboards that have been created are specific to the studies running them (with specific devices and requirements). These dashboards are added to the [Kubernetes installation](https://github.com/RADAR-base/RADAR-Kubernetes/tree/main/etc/radar-grafana/dashboards/allprojects). For example, one dashboard is called "Fitbit Weartime", which shows the daily number of hours the participant wore the Fitbit device. However, not all studies may be using the Fitbit device. It would be helpful to generalise the dashboards and allow easy configurations.
+
+**Dependencies:** Sample/Mock data will be necessary to create the graphs.
+
+**Goal:**
+
+| Goals | Related Issues | Requirements |
+|---------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| General Grafana dashboard configurations | - | - The goal is to create different Grafana dashboard configurations for different data/devices and types of visualisations needed. This would make dashboard setup easier and quicker for different studies. This would also allow easy reuse of configurations across studies.
+
+**Required Skills:** Grafana, TimescaleDB
```

Comment:
@Hsankesara, @mpgxvii  Maybe change this to SQL (as people might not be familiar with timescaledb)?